packages/editor/src/components/time-to-read/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { _x, _n } from '@wordpress/i18n';
+import { count as wordCount } from '@wordpress/wordcount';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+/**
+ * Average reading rate - based on average taken from
+ * https://irisreading.com/average-reading-speed-in-various-languages/
+ * (Characters/minute used for Chinese rather than words).
+ *
+ * @type {number} A rough estimate of the average reading rate across multiple languages.
+ */
+const AVERAGE_READING_RATE = 189;
+
+export default function TimeToRead() {
+	const content = useSelect(
+		( select ) => select( editorStore ).getEditedPostAttribute( 'content' ),
+		[]
+	);
+
+	/*
+	 * translators: If your word count is based on single characters (e.g. East Asian characters),
+	 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
+	 * Do not translate into your own language.
+	 */
+	const wordCountType = _x( 'words', 'Word count type. Do not translate!' );
+	const minutesToRead = Math.round(
+		wordCount( content, wordCountType ) / AVERAGE_READING_RATE
+	);
+
+	return (
+		<span className="time-to-read">
+			<span className="table-of-contents__number">
+				{ minutesToRead }{ ' ' }
```

Comment:
Is there a better way to leave a space between the number and the text, is CSS appropriate for this?

Reply:
There is also `createInterpolateElement` from `@wordpress/element` that can cover using React components:

https://github.com/WordPress/gutenberg/tree/trunk/packages/element#createinterpolateelement